### ISSUE TITLE
Event spawning_complete fires every time a user is created

### ISSUE THREAD

**Issue by NicoAdrian** [locustio/locust#2671]
### Prerequisites

- [X] I am using [the latest version of Locust](https://github.com/locustio/locust/releases/)
- [X] I am reporting a bug, not asking a question

### Description

The event spawning_complete is fired everytime a user is created inside a worker instead of when all users in the worker have spawned. Is it a bug or the intended behaviour ? If so, what can I do to fire a callback when all users inside a worker have spawned ? 

```python
@events.spawning_complete.add_listener
def on_spawning_complete(user_count):
    print("SPAWN", user_count)
```

Prints:
SPAWN 1
SPAWN 1
SPAWN 2
SPAWN 2
SPAWN 3

(I have 2 workers and set user count to 5 with a spawn rate of 1 in the GUI)

Instead, I expected only "SPAWN 3"

To add more details of what I want to do: I want to disable http decompression (there isn't a built-in option in the FastHTTPClient to do this, so I do `zlib.decompress = lambda *a, **b: None`, it's kinda hacky I know) once all users have started (after all users have completed their `on_start` method). 

### Command line

locust --worker

### Locustfile contents

```python3
@events.spawning_complete.add_listener
def on_spawning_complete(user_count):
    print("SPAWN", user_count)
```

### Python version

3.11

### Locust version

2.24.1

### Operating system

Linux CentOS

**Comment by cyberw** [comment]
It is kind of a bug. I think the issue was introduced in 2.0. It could be reimplemented on master side, but now workers dont really know when spawning has finished so it will be hard to make it work the way it used to.

**Comment by NicoAdrian** [comment]
@cyberw the event is correctly called on master side, when all users have spawned.

So I found a kinda ugly workaround:

1) On the `init` event, I register a message for workers (the callback passed will disable decompression) AND I get a global reference to the `environment` object.
2) On the `spawning_complete` event, I check if the runner is an instance of MasterRunner. If so, I sleep like 10s (to wait for all `on_start` method have completed), and then I send the message so the workers receive it and disable decompression

**Comment by NicoAdrian** [comment]
(Would be great if all the events receive the `environment` object, could be usefull)

**Comment by cyberw** [comment]
Well, when you say it like that, implementing spawning_completed on workers based on a message from master should be pretty easy :)

**Comment by NicoAdrian** [comment]
@cyberw Sorry I read let's NOT keep it open 😊

**Comment by cyberw** [comment]
Fixed in #2728
